src/App/App.tsx

```diff
@@ -39,56 +41,75 @@ const App: React.SFC = () => {
 	]);
   const [error, setError] = useState("");
 
+	useEffect(() => {getCocktail()}, []);
+	useEffect(() => {fetchAllCocktails()}, []);
+	useEffect(() => {
+		updateAllCocktails()
+	}, [allCocktails]);
+
   // API Calls
   const fetchAllCocktails = async (): Promise<any> => {
     try {
       const data: AllCocktailsDetails[] = await getAllCocktails();
-      return setAllCocktails(data);
+			return setAllCocktails(data);
     } catch (error) {
-      setError(error.message);
+      setAllCError(error.toString());
     }
   };
 
-	const getCocktail = async ():Promise<any> => {
+	const getCocktail = async ():Promise<void> => {
 		try {
 			const data: Cocktail = await getRandomCocktail();
-			setRandomCocktail(data);
+			return setRandomCocktail(data);
 		} catch (error) {
-			setError(error.message);
+			setRandomCError(error.toString());
 		}
 	};
 
-	useEffect(() => {getCocktail()}, []);
-	useEffect(() => {fetchAllCocktails()});
+	const updateAllCocktails = async ():Promise<void> => {
+		try {
+			const newCocktails = await Promise.all(
+				allCocktails.map(c => getCocktailDetails(c.idDrink))
+			);
+			setAllCocktails(newCocktails);
+		} catch (error) {
+			setError(error.message);
+		}
+	}
```

Comment:
Updates allCocktails w/ all extra props from the individual getCocktailDetails call